oshi-core/src/test/java/oshi/util/MemoizerTest.java

```diff
@@ -65,39 +67,59 @@ public void after() throws InterruptedException {
 
     @Test
     public void get() throws Throwable {
+        // Do 20+ seconds of tests with no refresh.
         long iterationDurationNanos = TimeUnit.MILLISECONDS.toNanos(1);
+        long ttlNanos = -1;
         for (int r = 0; r < 20_000; r++) {
-            assertTrue(run(iterationDurationNanos, -1));
+            run(iterationDurationNanos, ttlNanos);
         }
+        // Do 40+ seconds of tests with refresh after 0.1 ms.
         iterationDurationNanos = TimeUnit.MILLISECONDS.toNanos(10);
+        ttlNanos = iterationDurationNanos / 100;
+        assertNotEquals(0, ttlNanos); // avoid div/0 later
         for (int r = 0; r < 4000; r++) {
-            assertTrue(run(iterationDurationNanos, iterationDurationNanos / 100));
+            run(iterationDurationNanos, ttlNanos);
         }
     }
 
-    private boolean run(final long iterationDurationNanos, final long ttlNanos) throws Throwable {
+    private void run(final long iterationDurationNanos, final long ttlNanos) throws Throwable {
         final Supplier<Long> s = new Supplier<Long>() {
             private long value;
 
+            // this method is not thread-safe, the returned value of the counter may go down
+            // if this method is called concurrently from different threads
             @Override
-            public Long get() {// this method is not thread-safe, the counter may go down if MemoizedObject
-                                     // calls this method concurrently
+            public Long get() {
                 return ++value;
             }
         };
-        final Supplier<Long> m = Memoizer.memoize(s, ttlNanos);
+        // The memoizer we are testing
+        final Supplier<Long> m = memoize(s, ttlNanos);
+        // Hold the results until all threads terminate
         final Collection<Future<Void>> results = new ArrayList<>();
+        // Mark the start time, end after iterationDuration
         final long beginNanos = System.nanoTime();
         for (int tid = 0; tid < numberOfThreads; tid++) {
             results.add(ex.submit(() -> {
+                // First read from the memoizer. Only one thread will win this race but all
```

Comment:
Actually, `m.get()` calls here from each of the threads may lead to a refresh (a call to `s.get()` ) depending on the timings (though this does not change the max number of refreshes).

Reply:
Right. The first one is guaranteed to increment the value, the rest should at least see an incremented value (if not increment again if they are delayed until after ttl)

Reply:
The point is to note the test 2 lines lower that guarantees the >0 condition